### PR TITLE
Add machine output documentation

### PR DESCRIPTION
**Pull request by cdrage**
**What type of PR is this?**

> /kind documentation

**What does does this PR do / why we need it**:

Adds a table and references to machine output documentation

**Which issue(s) this PR fixes**:

N/A

**How to test changes / Special notes to the reviewer**:

View the doc

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

### CODE REVIEW

File: docs/dev/machine-output.adoc [hunk 1]
```diff
@@ -1,10 +1,173 @@
...
+| catalog
+| describe
+| N
+
+|
+| search
+| N
+
+|
+| <<catalog-list-components,list components>>
```
Comment (Preeticp) [inline docs/dev/machine-output.adoc]: This link and the ones on lines 35, 39, 111, 115, 131, 139 do not work. Most probably cause they are not  listed below, are we planning to add those sections here? Is that why we have these links?

File: docs/dev/machine-output.adoc [hunk 2]
```diff
@@ -133,12 +268,12 @@ This document outlines all the machine readable output options and examples.
     }
   ]
 }
-```
+----
 
+== component describe
```
Comment (Preeticp) [inline docs/dev/machine-output.adoc]: Component describe and Component list are described twice.

File: docs/dev/machine-output.adoc [hunk 3]
```diff
@@ -1,10 +1,173 @@
-# Machine Readable Output
+= Machine readable output
 
 This document outlines all the machine readable output options and examples.
 
-`odo app describe -o json | jq`
- 
-```json
+See the below table for a list of all possible machine readable output commands:
```
Comment (Preeticp) [inline docs/dev/machine-output.adoc]: @cdrage, iiuc, only those which have machine readable output implemented have the links? Do you think it makes sense to have the links to all of those which have sections outlined below?

Comment (cdrage) [reply docs/dev/machine-output.adoc]: Yup. Only the ones implemented have the links.

File: docs/dev/machine-output.adoc (hunk 1 above)
Comment (cdrage) [reply docs/dev/machine-output.adoc]: Yes, that's intentional, just wanted to see if this format is fine :+1:

**Pull request merged**